File: clients/java/src/main/java/com/svix/diom/apis/RateLimiterNamespace.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.RateLimiterCreateNamespaceIn;
import com.svix.diom.models.RateLimiterCreateNamespaceOut;
import com.svix.diom.models.RateLimiterGetNamespaceIn;
import com.svix.diom.models.RateLimiterGetNamespaceOut;

public class RateLimiterNamespace {
    private final HttpClient client;

    public RateLimiterNamespace(HttpClient client) {
        this.client = client;
    }

    /** Create rate limiter namespace */
    public RateLimiterCreateNamespaceOut create(
        final RateLimiterCreateNamespaceIn rateLimiterCreateNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limit/namespace/create");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterCreateNamespaceIn,
            RateLimiterCreateNamespaceOut.class
            );
    }

    /** Get rate limiter namespace */
    public RateLimiterGetNamespaceOut get(
        final RateLimiterGetNamespaceIn rateLimiterGetNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/rate-limit/namespace/get");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            rateLimiterGetNamespaceIn,
            RateLimiterGetNamespaceOut.class
            );
    }
}

File: clients/javascript/src/apis/rateLimiterNamespace.ts
// this file is @generated

import {
    type RateLimiterCreateNamespaceIn,
    RateLimiterCreateNamespaceInSerializer,
} from '../models/rateLimiterCreateNamespaceIn';
import {
    type RateLimiterCreateNamespaceOut,
    RateLimiterCreateNamespaceOutSerializer,
} from '../models/rateLimiterCreateNamespaceOut';
import {
    type RateLimiterGetNamespaceIn,
    RateLimiterGetNamespaceInSerializer,
} from '../models/rateLimiterGetNamespaceIn';
import {
    type RateLimiterGetNamespaceOut,
    RateLimiterGetNamespaceOutSerializer,
} from '../models/rateLimiterGetNamespaceOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class RateLimiterNamespace {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Create rate limiter namespace */
        public create(
            rateLimiterCreateNamespaceIn: RateLimiterCreateNamespaceIn,
            ): Promise<RateLimiterCreateNamespaceOut> {
            const request = new DiomRequest(HttpMethod.POST, "/api/v1/rate-limit/namespace/create");

            request.setBody(
                    RateLimiterCreateNamespaceInSerializer._toJsonObject(
                        rateLimiterCreateNamespaceIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    RateLimiterCreateNamespaceOutSerializer._fromJsonObject,
                );
            }

        

    /** Get rate limiter namespace */
        public get(
            rateLimiterGetNamespaceIn: RateLimiterGetNamespaceIn,
            ): Promise<RateLimiterGetNamespaceOut> {
            const request = new DiomRequest(HttpMethod.POST, "/api/v1/rate-limit/namespace/get");

            request.setBody(
                    RateLimiterGetNamespaceInSerializer._toJsonObject(
                        rateLimiterGetNamespaceIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    RateLimiterGetNamespaceOutSerializer._fromJsonObject,
                );
            }

        

    }

